inspirehep/modules/orcid/utils.py

```diff
@@ -236,12 +236,13 @@ def get_literature_recids_for_orcid(orcid):
 
     """
     orcid_object = '[{"schema": "ORCID", "value": "%s"}]' % orcid
-    author_recid = db.session.query(PersistentIdentifier.pid_value)\
-                             .filter(PersistentIdentifier.object_type == 'rec')\
-                             .filter(PersistentIdentifier.object_uuid == RecordMetadata.id)\
-                             .filter(PersistentIdentifier.pid_type == 'aut')\
-                             .filter(type_coerce(RecordMetadata.json, JSONB)['ids'].contains(orcid_object))\
-                             .one().pid_value
+    author_rec_uuid = db.session.query(RecordMetadata.id)\
+                    .filter(type_coerce(RecordMetadata.json, JSONB)['ids'].contains(orcid_object)).one().id
```

Comment:
Instead of `.one().id` -> `.scalar()`

Reply:
scalar has different semantics when there are no results: http://docs.sqlalchemy.org/en/latest/orm/query.html#sqlalchemy.orm.query.Query.scalar